src/rest_api/classes/transgene/widgets/expression.clj

```diff
@@ -0,0 +1,24 @@
+(ns rest-api.classes.transgene.widgets.expression
+  (:require
+    [clojure.string :as str]
+    [rest-api.formatters.object :as obj :refer [pack-obj]]
+    [rest-api.classes.expr-pattern.core :as expr-pattern]
+    [rest-api.classes.generic-functions :as generic-functions]
+    [rest-api.classes.generic-fields :as generic]))
+
+(defn marker-for [t]
+  {:data (some->> (:transgene/marker-for t)
+                  (first)
+                  (:transgene.marker-for/text))
+   :description "string decribing what the transgene is a marker for"})
+
+(defn expression-patterns [t]
+  {:data (some->> (:expr-pattern/_transgene t)
+                  (map (fn [ep]
```

Comment:
`map` returns an array, possibly empty one, and you probably need to use `seq` to convert it to `nil`

Reply:
Is fine because `some->>` will return nil if the last expression is empty or nil.